pilot/pkg/networking/plugin/authz/util_test.go

```diff
@@ -209,6 +209,31 @@ func TestConvertToHeaderMatcher(t *testing.T) {
 	}
 }
 
+func TestIsKeyBinary(t *testing.T) {
+	cases := []struct {
+		s      string
+		expect bool
+	}{
+		{s: "a[b]", expect: true},
+		{s: "a", expect: false},
+		{s: "a.b", expect: false},
+		{s: "a.b[c]", expect: true},
+		{s: "a.b[c.d]", expect: true},
+		{s: "[a]", expect: false},
+		{s: "[a", expect: false},
+		{s: "a]", expect: false},
+		{s: "a[]", expect: false},
+		{s: "a.b[c.d]e", expect: false},
+		{s: "a.b[[c.d]]", expect: true},
```

Comment:
Is `a.b[[c.d]]` also considered valid?

Reply:
I think it's okay to have a value like `[c.d]` for a list matcher.